src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -414,36 +406,11 @@ static void CheckForExperimentStatus(ITextView textView, in CompletionOptions op
         /// to transient objects, which would cause memory leak (among other potential issues) if cached. 
         /// So as a compromise,  we cache data that can be calculated from Roslyn completion item to avoid repeated 
         /// calculation cost for cached Roslyn completion items.
+        /// FilterSetData is the bit vector value from the FilterSet of this item.
         /// </summary>
-        private readonly struct VSCompletionItemData
-        {
-            public VSCompletionItemData(
-                string displayText, ImageElement icon, ImmutableArray<AsyncCompletionData.CompletionFilter> filters,
-                int filterSetData, ImmutableArray<ImageElement> attributeIcons, string insertionText)
-            {
-                DisplayText = displayText;
-                Icon = icon;
-                Filters = filters;
-                FilterSetData = filterSetData;
-                AttributeIcons = attributeIcons;
-                InsertionText = insertionText;
-            }
-
-            public string DisplayText { get; }
-
-            public ImageElement Icon { get; }
-
-            public ImmutableArray<AsyncCompletionData.CompletionFilter> Filters { get; }
-
-            /// <summary>
-            /// This is the bit vector value from the FilterSet of this item.
-            /// </summary>
-            public int FilterSetData { get; }
-
-            public ImmutableArray<ImageElement> AttributeIcons { get; }
-
-            public string InsertionText { get; }
-        }
+        private readonly record struct VSCompletionItemData
+            (string DisplayText, ImageElement Icon, ImmutableArray<AsyncCompletionData.CompletionFilter> Filters,
+                int FilterSetData, ImmutableArray<ImageElement> AttributeIcons, string InsertionText);
```

Comment:
move to separate file.  my pref on formatting would be:

```
        private readonly record struct VSCompletionItemData(
            string DisplayText,
            ImageElement Icon,
            ImmutableArray<AsyncCompletionData.CompletionFilter> Filters,
            int FilterSetData,
            ImmutableArray<ImageElement> AttributeIcons,
            string InsertionText);
```

Taht way all the properties are very clear.